src/routes.rs

```diff
@@ -666,14 +666,21 @@ fn get_catalog(
     }
 }
 
-#[get("/v2/<repo_name>/tags/list")]
+#[get("/v2/<repo_name>/tags/list?<limit>")]
 fn list_tags(
     _auth_user: TrowToken,
     ci: rocket::State<ClientInterface>,
     repo_name: String,
+    limit: Option<i32>
 ) -> Result<TagList, Error> {
     let rn = RepoName(repo_name);
-    let tags = ci.list_tags(&rn);
+    let mut lim = std::i32::MAX;
+    match limit {
+        Some(i) => lim = i,
+        None => {}
```

Comment:
I guess you could do
```
let limit = match limit {
  Some(l) => l,
  None => std::i32::MAX
}
```
Also std::i32::MAX seems quite a lot of results? Should we have a lower default value?